financial_transcripts/preprocessing/generate_benchmark_qa.ipynb

```diff
@@ -780,12 +741,109 @@
     "print(f\"Data has been saved to {csv_file_path}.\")"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "AI as a Judge: Adding a superior AI model as a judge to evaluate the generated questions and answers. "
+   ]
+  },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 24,
    "metadata": {},
    "outputs": [],
-   "source": []
+   "source": [
+    "def get_ai_similarity(question_answer_set, chunk1 = chunk1, chunk2 = chunk2):\n",
+    "    \n",
+    "    \n",
+    "    eval_prompt= f\"\"\" For the given set of questions and answers, determine if the answer is relevant to either of the contexts.\n",
+    "    Set of questions and answers: {question_answer_set}\n",
+    "    Context1: {chunk1}\n",
+    "    Context2: {chunk2}\n",
+    "    Evaluate the accuracy of the question answer extraction model.\n",
+    "    Given a set of 10 questions and answers, the context sources Context1 and Context2,  determine if the model's question and answer is relevant to one of the contexts.\n",
+    "    Return only a single score of 0 or 1 indicating whether the model's question and answer is grounded in either of the contexts. Rewrite each of the questions and answers, and add a line for score you give. For example:\n",
```

Comment:
if i were a human who got this instruction, i would be confused :) first you ask to 'determine if the answer is relevant to either of the context', then you ask to 'determine if the model's _question and_ answer  is relevant to one of the contexts. You also reformulate as 'evaluate the accuracy of the question answer extraction model' (formally speaking, we don't have extraction here - because extraction suggests taking a few pieces from source text as-is, without any changes to words or order). Then another reformulation, 'Return only a single score of 0 or 1 indicating  whether the model's question and answer is grounded in either of the contexts'. maybe we can use GPT to summarize the prompt? :) 

Also, traditionally AI assisted metrics are on the scale of 1 to 5. If you ask to evaluate both question and answers, and question ideally needs to be based on two chunks, maybe define scale as follows: 
5 - question is good, information from both chunks is necessary to answer it
4 - question is good, information from only one chunk is sufficient to answer it
3 - question is grounded on information from the chunks, but it's not good question (e.g. not clear what is asked)
2 - question is good, but not based on information from the chunks
1 - question is bad and not based on information from the chunks

I also think answer may need to be evaluated separately - e.g. is it comprehensive? Based on feedback we got from financial experts, a lot of them weren't

Reply:
Thanks, this scaling is good to experiment with. The metric 1-5 is mainly for AI Studio metrics they decided, 0 or 1 is more a binary metric, and standard quick checking. But, it does not have to be that way alone. Should experimentation of prompts be part of the notebook? I thought and intended it is more a representative approach, showing a path for other use cases, rather than solving a use-case specific refined approach and then providing a specific solution. Experimentation of prompts can be another separate notebook also, if that is suggested.

Reply:
I will add a prompt template saving prompts for experimentation and include these updates to prompts also.

Reply:
Updated in commits below, all we discussed.